build-tools/packages/build-tools/src/typeValidator/packageJson.ts

```diff
@@ -464,25 +479,34 @@ export async function getAndUpdatePackageDetails(
 		packageDetails.json.devDependencies = createSortedObject(
 			packageDetails.json.devDependencies,
 		);
-		const disabled = packageDetails.json.typeValidation?.disabled;
+		const disabled = packageDetails.typeValidation?.disabled;
 
-		packageDetails.json.typeValidation = {
+		packageDetails.typeValidation = {
 			version,
 			previousVersionStyle,
 			baselineRange: baseline,
 			baselineVersion: baseline === prevVersion ? undefined : prevVersion,
-			broken: resetBroken === true ? {} : packageDetails.json.typeValidation?.broken ?? {},
+			broken: resetBroken === true ? {} : packageDetails.typeValidation.broken ?? {},
 		};
 
 		if (disabled !== undefined) {
-			packageDetails.json.typeValidation.disabled = disabled;
+			packageDetails.typeValidation.disabled = disabled;
 		}
 
 		if ((writeUpdates ?? false) === true) {
-			await util.promisify(fs.writeFile)(
-				`${packageDir}/package.json`,
-				JSON.stringify(packageDetails.json, undefined, 2).concat("\n"),
+			await writeJson(
+				`${packageDir}/typeValidation.config.json`,
```

Comment:
Nit: Seems like most json config files are named using kebab case (package-lock, api-extractor, etc.). Should we considfer `type-validation.config.json`?

Reply:
Done in latest.